lib/core/utils/qsa.js

```diff
@@ -6,32 +6,27 @@ var matchExpressions = function() {};
 
 function matchesTag(node, exp) {
 	return (
-		node.nodeType === 1 &&
-		(exp.tag === '*' || node.nodeName.toLowerCase() === exp.tag)
+		node.elementNodeType === 1 &&
+		(exp.tag === '*' || node.elementNodeName === exp.tag)
 	);
 }
 
 function matchesClasses(node, exp) {
-	return (
-		!exp.classes ||
-		exp.classes.reduce((result, cl) => {
-			return result && (node.className && node.className.match(cl.regexp));
-		}, true)
-	);
+	return !exp.classes || exp.classes.every(cl => node.hasClass(cl.value));
 }
 
 function matchesAttributes(node, exp) {
 	return (
 		!exp.attributes ||
 		exp.attributes.reduce((result, att) => {
-			var nodeAtt = node.getAttribute(att.key);
+			var nodeAtt = node.attr(att.key);
```

Comment:
The following line could raise an exception.

Reply:
None that it wouldn't have raised before